src/java.base/share/classes/com/sun/crypto/provider/SunJCE.java

```diff
@@ -339,11 +357,23 @@ public Object run() {
                     put("Cipher.ARCFOUR SupportedPaddings", "NOPADDING");
                     put("Cipher.ARCFOUR SupportedKeyFormats", "RAW");
 
-                    put("Cipher.ChaCha20",
-                        "com.sun.crypto.provider.ChaCha20Cipher$ChaCha20Only");
+                    if (useNativeChaCha20Cipher) {
+                        put("Cipher.ChaCha20",
+                                "com.sun.crypto.provider.NativeChaCha20Cipher$ChaCha20Only");
+                        put("Cipher.ChaCha20-Poly1305",
+                                "com.sun.crypto.provider.NativeChaCha20Cipher$ChaCha20Poly1305");
+                    } else {
+                        put("Cipher.ChaCha20",
+                                "com.sun.crypto.provider.ChaCha20Cipher$ChaCha20Only");
+                        put("Cipher.ChaCha20-Poly1305",
+                                "com.sun.crypto.provider.ChaCha20Cipher$ChaCha20Poly1305");
+                    }
+
+
+
```

Comment:
Please remove the rendundant blank lines (we don't need two or more together).

Reply:
fixed